exe/task.cc

```diff
@@ -53,6 +55,56 @@ static string system_prefix()
 string test_prefix();
 #endif
 
+task task_runner::find_task(string taskname) {
+    string module, task = "init";
+    if (!lth_util::re_search(taskname, boost::regex("\\A(\\w+)\\z"), &module) &&
+        !lth_util::re_search(taskname, boost::regex("\\A(\\w+)::(\\w+)\\z"), &module, &task)) {
+        throw task_error("invalid-task", _("Invalid task name '{1}'", taskname));
+    }
+
+    auto filepath = SYSTEM_PREFIX()+"/pxp-agent/tasks/"+module+"/tasks";
+
+    // Search for any executable file starting with the task name and not matching a set of reserved extensions
+    vector<string> filenames;
+    string extension;
+
+    try {
+        fs::directory_iterator end;
+        for (auto f = fs::directory_iterator(filepath); f != end; ++f) {
+            if (!fs::is_directory(f->status()) && f->path().stem() == task) {
+                extension = f->path().extension().string();
+                if (reserved_extensions_.find(extension) != reserved_extensions_.end()) {
+                    continue;
+                }
+
+                filenames.emplace_back(f->path().string());
+            }
+        }
+    } catch (fs::filesystem_error &e) {
+        // Fall through to filenames.empty
+    }
+
+    if (filenames.empty()) {
+        throw task_error("not-found", _("Could not find task '{1}' at {2}", taskname, filepath));
+    }
+    if (filenames.size() > 1) {
+        throw task_error("multiple-files", _("Found multiple matching files for task '{1}': {2}", taskname,
+                                             boost::algorithm::join(filenames, ", ")));
+    }
+
+    auto filename = filenames.front();
+
+    auto builtin = builtin_tasks_.find(extension);
+    if (builtin != builtin_tasks_.end()) {
+        return builtin->second(filename);
+    }
+
+    if (lth_exec::which(filename).empty()) {
```

Comment:
No. For .bat, .cmd, .exe, or .com it will return true. And seems to execute at least .bat and .exe correctly.